engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -566,6 +577,10 @@ class WorkflowActor(workflowToStart: WorkflowToStart,
         }
       }
 
+      // Attempt to perform workflow completion callback
+      val callbackUri = workflowOptions.get(WorkflowOptions.WorkflowCallbackUri).toOption
+      self ! PerformWorkflowCallback(callbackUri, terminalState.workflowState)
```

Comment:
I initially implemented this as part of finalization (you can see that working in earlier commits) but that turned out not to work well. Workflows are still Running, and may still fail, during finalization. Also, if a workflow fails very early in its lifecycle we may not ever perform finalization. Putting it here, at the moment of transition to terminal state, seemed right. 

I wanted to send a message directly to the `WorkflowCallbackActor` from here, but couldn't find a way to get access to the final workflow outputs. That's why I send an event to this actor, and get them from the `data` where the event is handled. If there's a way to get the actor data here I could simplify this a little.